common/src/main/java/tachyon/security/group/UserToGroupsMappingService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.group;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.CommonUtils;
+
+/**
+ * A user-to-groups mapping service.
+ *
+ * {@link UserToGroupsMappingService} allows for server to get the various group memberships of a
+ * given user via the {@link #getGroups(String)} call, thus ensuring a consistent user-to-groups
+ * mapping and protects against mapping inconsistencies between servers and clients in a Tachyon
+ * cluster.
+ */
+public final class UserToGroupsMappingService {
```

Comment:
Thank @yupeng9 , **GroupMappingService** is an interface, it will have different implementations for different cases. And **UserToGroupsMappingService** (Groups) will handle the work like initializing GroupMappingService. We may need **GroupsMappingService** at many places, we can use **UserToGroupsMappingService** (Groups) to **getGroups** directly. Briefly,  **GroupMappingService** is an interface for service,  **UserToGroupsMappingService**(Groups) is used by client.